preview/MsixCore/msixmgrLib/PackageManager.cpp

```diff
@@ -180,20 +198,30 @@ HRESULT PackageManager::FindPackageByFamilyName(const wstring & packageFamilyNam
     RETURN_IF_FAILED(filemapping.GetInitializationResult());
     auto msixCoreDirectory = filemapping.GetMsixCoreDirectory();
 
+    wstring packageFamilyNameCopy = packageFamilyName;
+
+    packageFamilyNameCopy = std::regex_replace(packageFamilyNameCopy, std::wregex(L"\\*"), L".*");
+    packageFamilyNameCopy = std::regex_replace(packageFamilyNameCopy, std::wregex(L"\\?"), L".");
+
+    std::string packageFamilyNameString(packageFamilyNameCopy.begin(), packageFamilyNameCopy.end());
+    std::regex packageFamilyNameRegExp(packageFamilyNameString);
+
     for (auto& p : experimental::filesystem::directory_iterator(msixCoreDirectory))
     {
         if (experimental::filesystem::is_directory(p.path()))
         {
-            auto installedAppFamilyName = GetFamilyNameFromFullName(p.path().filename());
-            if (CaseInsensitiveEquals(installedAppFamilyName, packageFamilyName))
+            wstring installedAppFamilyName = GetFamilyNameFromFullName(p.path().filename());
+            std::string installedAppFamilyNameString(installedAppFamilyName.begin(), installedAppFamilyName.end());
+
+            if (std::regex_match(installedAppFamilyNameString, packageFamilyNameRegExp))
             {
                 wstring packageDirectoryPath = msixCoreDirectory + std::wstring(p.path().filename());
                 RETURN_IF_FAILED(GetPackageInfo(packageDirectoryPath, installedPackage));
                 return S_OK;
             }
         }
     }
-    return S_OK;
+    return ERROR_NOT_FOUND;

```

Comment:
>ERROR_NOT_FOUND [](start = 11, length = 15)

should be HRESULT_FROM_WIN32(ERROR_NOT_FOUND) as the return type of this should be HRESULT.

Reply:
Modified to HRESULT_FROM_WIN32(ERROR_NOT_FOUND).